File: frontend/src/renderer/src/api/operations.ts
import client from "./client";

export interface Operation {
  id: number;
  part_id: number;
  sequence_no: number;
  operation_type: string;
  machine_name?: string;
  tool_name?: string;
  setup_time_min: number;
  cycle_time_min: number;
  machine_rate_hr: number;
  tool_cost: number;
  notes?: string;
  created_at: string;
  machining_cost: number;
}

export async function getOperations(part_id?: number): Promise<Operation[]> {
  const res = await client.get("/operations", { params: { part_id } });
  return res.data;
}

export async function createOperation(
  data: Omit<Operation, "id" | "created_at" | "machining_cost">
): Promise<Operation> {
  const res = await client.post("/operations", data);
  return res.data;
}

export async function deleteOperation(id: number): Promise<void> {
  await client.delete(`/operations/${id}`);
}

File: backend/app/routers/operations.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from app.database import get_db
from app.models.operation import Operation

router = APIRouter(prefix="/operations", tags=["operations"])


class OperationCreate(BaseModel):
    part_id: int
    sequence_no: int
    operation_type: str
    machine_name: Optional[str] = None
    tool_name: Optional[str] = None
    setup_time_min: float = 0.0
    cycle_time_min: float
    machine_rate_hr: float
    tool_cost: float = 0.0
    notes: Optional[str] = None


class OperationResponse(OperationCreate):
    id: int
    created_at: datetime
    machining_cost: float

    class Config:
        from_attributes = True


@router.get("/", response_model=list[OperationResponse])
def get_operations(part_id: Optional[int] = None, db: Session = Depends(get_db)):
    query = db.query(Operation)
    if part_id:
        query = query.filter(Operation.part_id == part_id)
    return query.order_by(Operation.sequence_no).all()


@router.post("/", response_model=OperationResponse)
def create_operation(op: OperationCreate, db: Session = Depends(get_db)):
    db_op = Operation(**op.model_dump())
    db.add(db_op)
    db.commit()
    db.refresh(db_op)
    return db_op


@router.delete("/{op_id}")
def delete_operation(op_id: int, db: Session = Depends(get_db)):
    op = db.query(Operation).filter(Operation.id == op_id).first()
    if not op:
        raise HTTPException(status_code=404, detail="Operasyon bulunamadı")
    db.delete(op)
    db.commit()
    return {"message": "Silindi"}
